### ISSUE TITLE
Verify requirements.txt contents for 0.13 branch 

### ISSUE THREAD

**Issue by simaishi** [pulp/pulp_ansible#1102]
Trying to build pulp_ansible rpm from 0.13 branch head and noticed some of the dependencies specified in 0.13 could cause downgrade when upgrading to a later pulp_ansible version. As per discussion with @fao89, opening an issue to verify the requirements.

* async_lru - 0.13 branch allows any 1.x, while 0.14 allows 1.0.x only
  * 0.13 branch: ~=1.0
  * 0.14 branch: >=1.0,<1.1
* jsonschema - 0.13 branch allows any 4.x with minimum 4.4, while 0.14 allows up to 4.7
  * 0.13 branch: ~=4.4
  * 0.14 branch: >=4.4,<4.7
* packaging - 0.13 needs minimum 21.3, while 0.14 can use any 21.x
  * 0.13 branch: ~=21.3
  * 0.14 branch: >=21,<22
* semantic_version - 0.13 needs minimum 2.10 and any 2.x is allowed, while 0.14 allows 2.9 to 2.11
  * 0.13 branch: ~=2.10
  * 0.14 branch: >=2.9,<2.11

**Comment by gerrod3** [comment]
Yes this is a recent change we have decided on because using ~= was both too permissive and restricted since our dependencies don't all follow semantic versioning causing our pins to be inconsistent. For 0.14 and pulpcore 3.20+ (and our other plugins) we are switching over to using ranges of known compatible versions and have dependabot update the upper bound accordingly when new releases come out. Hopefully this will reduce future incompatibility errors.

As for how 0.14 requirements got decided, I went through each dependency's release history and checked if they used semantic versioning to determine how permissive of a range we can allow. That's why some of the requirements are now more restrictive, to ensure future dependency releases won't break future 0.14 installs.
